### DISCUSSION TITLE
Can someone write me a simple make file for g++

### DISCUSSION THREAD

**Discussion by kijetesantakalu042** [community/community#136322] [Programming Help]
### Body

I have a cpp project. The directory looks like this:
- Root
   - Build
      - .o files
   - Source
      - Source (.cpp .hpp) files
   - Makefile

I need to make a Makefile that will build files in Source and link them.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Haseebasif7** [comment]
Use GPT:

# Compiler
CXX = g++

# Compiler flags
CXXFLAGS = -Wall -Wextra -std=c++17

# Linker flags (if any)
LDFLAGS =

# Directories
SRCDIR = Source
BUILDDIR = Build

# Find all source files
SRC = $(wildcard $(SRCDIR)/*.cpp)

# Find all object files
OBJ = $(patsubst $(SRCDIR)/%.cpp,$(BUILDDIR)/%.o,$(SRC))

# Executable name
TARGET = my_program

# Default target
all: $(TARGET)

# Link the executable
$(TARGET): $(OBJ)
	$(CXX) $(LDFLAGS) -o $@ $^

# Compile source files into object files
$(BUILDDIR)/%.o: $(SRCDIR)/%.cpp
	@mkdir -p $(BUILDDIR)
	$(CXX) $(CXXFLAGS) -c $< -o $@

# Clean up build files
clean:
	rm -f $(BUILDDIR)/*.o $(TARGET)

# Phony targets
.PHONY: all clean
